### PR TITLE
Add support for receiving sendaddrv2 message from a peer

### LINKED ISSUE #1661: Btcd disconnects upon receiving sendaddrv2
Bitcoin Core master, soon to be released, introduces the new `sendaddrv2` message ([BIP 155](https://github.com/bitcoin/bips/blob/master/bip-0155.mediawiki)). When such a Bitcoin Core node connects to a btcd node (tag v0.21.0-beta) it immediately disconnects:

```
2020-11-12 18:31:13.935 [ERR] PEER: Can't read message from x.x.x.x:42192 (inbound): ReadMessage: unhandled command [sendaddrv2]
2020-11-12 18:31:13.933 [INF] SYNC: New valid peer x.x.x.x:42192 (inbound) (/Satoshi:0.20.99/)
2020-11-12 18:31:13.946 [INF] SYNC: Syncing to block height 656631 from peer x.x.x.x:42192
2020-11-12 18:31:13.946 [INF] SYNC: Lost peer x.x.x.x:42192 (inbound)
```

On bitcoind side:

```
2020-11-12T18:31:13Z Added connection peer=6606
2020-11-12T18:31:13Z sending version (103 bytes) peer=6606
2020-11-12T18:31:13Z send version message: version 70016, blocks=656631, us=[::]:0, peer=6606
2020-11-12T18:31:13Z received: version (113 bytes) peer=6606
2020-11-12T18:31:13Z sending verack (0 bytes) peer=6606
2020-11-12T18:31:13Z sending sendaddrv2 (0 bytes) peer=6606
2020-11-12T18:31:13Z sending getaddr (0 bytes) peer=6606
2020-11-12T18:31:13Z receive version message: /btcwire:0.5.0/btcd:0.21.0/: version 70013, blocks=656631, us=x.x.x.x:42192, peer=6606
2020-11-12T18:31:13Z received: verack (0 bytes) peer=6606
2020-11-12T18:31:13Z New outbound peer connected: version: 70013, blocks=656631, peer=6606 (full-relay)
2020-11-12T18:31:13Z sending sendheaders (0 bytes) peer=6606
2020-11-12T18:31:13Z sending ping (8 bytes) peer=6606
2020-11-12T18:31:13Z sending addr (31 bytes) peer=6606
2020-11-12T18:31:13Z initial getheaders (656630) to peer=6606 (startheight:656631)
2020-11-12T18:31:13Z sending getheaders (1029 bytes) peer=6606
2020-11-12T18:31:13Z sending feefilter (8 bytes) peer=6606
2020-11-12T18:31:13Z received: reject (110 bytes) peer=6606
2020-11-12T18:31:13Z Unknown command "reject" from peer=6606
2020-11-12T18:31:13Z socket recv error for peer=6606: Connection reset by peer (104)
2020-11-12T18:31:13Z disconnecting peer=6606
2020-11-12T18:31:13Z Cleared nodestate for peer=6606
```

I tried with and without the new `peerblockfilters` feature, but that doesn't seem to matter.

This may lead to node loneliness. It also happens to Lnd nodes, see https://github.com/breez/breezmobile/issues/408

This issue did not occur as of a slightly older Bitcoin Core commit, e.g. https://github.com/bitcoin/bitcoin/commit/1d3ec2a1fda7446323786a52da1fd109c01aa6fb

### PR DESCRIPTION
**Pull request by yaslama**
Fixes #1661

### CODE REVIEW

File: wire/message.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ const (
 	CmdCFilter      = "cfilter"
```
Comment (shahriyar9839) [inline wire/message.go:57]: 0x915133E981cD5D9616F191977a8896522d56Cd7C

File: wire/msgsendaddrv2.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package wire
+
+import (
+	"io"
+)
+
+// MsgSendAddrV2 defines a bitcoin verack message which is used for a peer to
+// signals support for receiving ADDRV2 messages (BIP155). It implements the
+// the Message interface.
+//
```
Comment (shahriyar9839) [inline wire/msgsendaddrv2.go:10]: 0x915133E981cD5D9616F191977a8896522d56Cd7C

File: wire/msgsendaddrv2.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+// This message has no payload.
+type MsgSendAddrV2 struct{}
+
+// BtcDecode decodes r using the bitcoin protocol encoding into the receiver.
+// This is part of the Message interface implementation.
+func (msg *MsgSendAddrV2) BtcDecode(r io.Reader, pver uint32, enc MessageEncoding) error {
```
Comment (shahriyar9839) [inline wire/msgsendaddrv2.go:16]: 0x915133E981cD5D9616F191977a8896522d56Cd7C

File: wire/msgsendaddrv2.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	return nil
+}
+
+// BtcEncode encodes the receiver to w using the bitcoin protocol encoding.
```
Comment (shahriyar9839) [inline wire/msgsendaddrv2.go:20]: 0x915133E981cD5D9616F191977a8896522d56Cd7C

**Review by shahriyar9839 — COMMENTED** [review]
0x915133E981cD5D9616F191977a8896522d56Cd7C

File: wire/msgsendaddrv2.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
+package wire
+
+import (
+	"io"
+)
+
+// MsgSendAddrV2 defines a bitcoin verack message which is used for a peer to
```
Comment (Roasbeef) [inline wire/msgsendaddrv2.go:7]: verack message?

File: wire/msgsendaddrv2.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+// signals support for receiving ADDRV2 messages (BIP155). It implements the
```
Comment (Roasbeef) [inline wire/msgsendaddrv2.go:8]: signals -> signal

**Review by Roasbeef — COMMENTED** [review]
Two small comment nits, but other than that should resolve the current issue w.r.t neutrino peers being able to connect to `bitcoind` master, and their upcoming release as well. I still wonder why a protocol version bump wasn't used instead of introducing a new message, but hey we're already here 🤷‍♂️

**Review by jakesyl — APPROVED** [review]
Looks good to me

**Review by Roasbeef — APPROVED** [review]
LGTM 🥟

Will update the `neutrino` project as well so the main package in isolation (when not used with `lnd` which will also pull in this change) can speak with the latest `bitcoind` nodes.

**Pull request merged**